pkg/commands/docker.go

```diff
@@ -144,12 +145,40 @@ func PopulateDigests(runner build.Runner, dependencies []build.ImageDependencies
 
 func queryDigest(runner build.Runner, reference *build.ImageReference) error {
 	refString := reference.String()
-	line, err := runner.QueryCmd("docker", []string{
-		"image", "ls", "--digests", "--format", "\"{{.Digest}}\"", refString,
+	output, err := runner.QueryCmd("docker", []string{
+		"inspect", "--format", "\"{{json .RepoDigests}}\"", refString,
 	})
 	if err != nil {
 		return err
 	}
-	reference.Digest = strings.TrimSpace(line)
+
+	trimCharPredicate := func(c rune) bool {
+		return '\n' == c || '\r' == c || '"' == c || '\t' == c
+	}
+
+	reference.Digest = getRepoDigest(strings.TrimFunc(output, trimCharPredicate), reference)
 	return nil
 }
+
+func getRepoDigest(jsonContent string, reference *build.ImageReference) string {
+	// Input: ["docker@sha256:b90307d28c6a6ab3d1d873d03a26c53c282bb94d5b5fb62cc7c027c384fe50ce"], , docker
+	// Output: sha256:b90307d28c6a6ab3d1d873d03a26c53c282bb94d5b5fb62cc7c027c384fe50ce
+
+	// Input: ["test.azurecr.io/docker@sha256:b90307d28c6a6ab3d1d873d03a26c53c282bb94d5b5fb62cc7c027c384fe50ce"], test.azurecr.io, docker
+	// Output: sha256:b90307d28c6a6ab3d1d873d03a26c53c282bb94d5b5fb62cc7c027c384fe50ce
+
+	prefix := reference.Repository + "@"
+	if len(reference.Registry) > 0 {
+		prefix = reference.Registry + "/" + prefix
```

Comment:
As discussed earlier, multi-arch manifest digest would appear here mostly for docker hub official images. Assuming `reference.Registry` is an ACR registry, most likely the digest obtained here is for the specific platform.

Reply:
Thanks. For multi-arch digest, I opened #57 to track it.